test/entities/swap.test.ts

```diff
@@ -0,0 +1,128 @@
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[1].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[1].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[1].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+            weightedPoolFactory({
+                id: '0xcb444e90d8198415266c6a2724b7900fb12fc56e000000000000000000000011' as Hex,
+                tokens: [
+                    {
+                        address: tokens[1].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[1].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[1].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+            weightedPoolFactory({
+                id: '0x177127622c4a00f3d409b75571e12cb3c8973d3c000000000000000000000011' as Hex,
+                tokens: [
+                    {
+                        address: tokens[2].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[2].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[2].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 0,
+                    },
+                    {
+                        address: tokens[3].address,
+                        weight: '0.5',
+                        balance: String(
+                            parseUnits('100000', tokens[3].decimals),
+                        ) as `${number}`,
+                        decimals: tokens[3].decimals,
+                        symbol: 'XXX',
+                        name: 'XXX Coin',
+                        index: 1,
+                    },
+                ],
+            }),
+        ];
+
+        test('amountIn to be maximally 10% more then expected', () => {
```

Comment:
Should be 1% based on the value you're testing, right?

Reply:
exactly, good catch!